components/prize_pool/mobilelegends/prize_pool_custom.lua

```diff
@@ -0,0 +1,64 @@
+---
+-- @Liquipedia
+-- wiki=mobilelegends
+-- page=Module:PrizePool/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Arguments = require('Module:Arguments')
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Variables = require('Module:Variables')
+
+local PrizePool = Lua.import('Module:PrizePool', {requireDevIfEnabled = true})
+
+local LpdbInjector = Lua.import('Module:Lpdb/Injector', {requireDevIfEnabled = true})
+local CustomLpdbInjector = Class.new(LpdbInjector)
+
+local CustomPrizePool = {}
+
+local TIER_VALUE = {32, 16, 8, 4, 2}
+
+-- Template entry point
+function CustomPrizePool.run(frame)
+	local args = Arguments.getArgs(frame)
+	local prizePool = PrizePool(args):create()
+
+	prizePool:setLpdbInjector(CustomLpdbInjector())
+
+	return prizePool:build()
+end
+
+function CustomLpdbInjector:adjust(lpdbData, placement, opponent)
+	lpdbData.weight = CustomPrizePool.calculateWeight(
+		lpdbData.prizemoney,
+		Variables.varDefault('tournament_liquipediatier'),
+		placement.placeStart
+	)
+
+	lpdbData.publishertier = Variables.varDefault('tournament_publishertier', '')
+	lpdbData.extradata.publisherpremier = Variables.varDefault('tournament_publisher_major') and 'true' or ''
+
+	local team = lpdbData.participant or ''
+	local smwPrefix = Variables.varDefault('smw_prefix', '')
```

Comment:
not a fan of this, it will miss it if the prefix is entered in the prizepool directly
(the old ppt set the var in the start template so it was available during storage, this is not the case with the new one)
you could directly grab it from `_prizePool.options.lpdbPrefix`
if you set `_prizePool = prizepool` at line 28

additionally imo rename to lpdbPrefix

Reply:
Got it, were the steps is to add that at line 28 and rename line 45 to something like this? (if I assume it correctly) 
![image](https://user-images.githubusercontent.com/88981446/188851865-5de03e66-78cb-4e41-abd5-0c36bc2bea59.png)

(or its just add 28 and remove 45?)

Reply:
line 23:
```lua
local _prize_pool

```
line 28:
```lua
	_prize_pool = prizepool

```

```suggestion
	local lpdbPrefix = _prize_pool.options.lpdbPrefix
```

Reply:
ok, have added all those but currently it will returned error at line 45
**attempt to index upvalue '_prize_pool' (a nil value).**

The added revision is now on git for checking

Reply:
adjusted it for you